File: Python/12(Highly divisible triangular number)_D5.py
import math
cycle = 0
currentNum = 1
currentCycle = 1
divisors = 0
while cycle == 0:
    for x in range(1, int(math.ceil(math.sqrt(currentNum)))):
        if currentNum % x == 0:
            divisors +=2
    print(f'Current number: {currentNum} Number of Divisors: {divisors}')
    if divisors > 500:
        cycle +=1
    else:
        currentCycle +=1
        currentNum = currentNum + currentCycle
        divisors = 0

File: Javascript/12(Highly divisible triangular number)_D5.js
var cycle = 1;
var currentNumber = 1;
var highestNum = 0;
for(highestDivisor = 0;highestDivisor <500;){
    var calcNumber = Math.floor(Math.sqrt(currentNumber));
    var divisor = 2;
    if(calcNumber > 1){
        for(x = 2; x <=calcNumber; x++){
            if(currentNumber % x == 0){
                divisor +=2;
            }
        }
    }
    if(divisor > highestDivisor){
        highestDivisor = divisor;
        highestNum = currentNumber
    }
    var nextNumber = currentNumber + cycle + 1;
    currentNumber = nextNumber;
    cycle++;
}
console.log(highestNum);